pkg/stub/config.go

```diff
@@ -67,94 +67,93 @@ func (h *Handler) SpecValidation(cfg *v1.Config) error {
-			afterError := cfg.NameInReason(importErrors.Reason, stream)
-			if beforeError && !afterError {
-				clearImageImportErrors = true
-				// we do not break here cause we want to clear out all possible streams
-			}
-		}
-		return true, false, clearImageImportErrors
+		configChangeAtAll = true
+		streamChange = true
 	}
 
-	clearImageImportErrors := false
-	changeInContent := false
-	for _, skip := range cfg.Spec.SkippedImagestreams {
-		if _, ok := streamsThatWereSkipped[skip]; !ok {
-			logrus.Printf("imagestream %s no longer skipped", skip)
-			changeInContent = true
-			importErrors := cfg.Condition(v1.ImportImageErrorsExist)
-			beforeError := cfg.NameInReason(importErrors.Reason, skip)
-			importErrors = h.clearStreamFromImportError(skip, importErrors, cfg)
-			afterError := cfg.NameInReason(importErrors.Reason, skip)
-			if beforeError && !afterError {
-				clearImageImportErrors = true
-			}
+	streamsThatWereSkipped := map[string]bool{}
+	unskippedStreams := map[string]bool{}
+	// build the list of skipped imagestreams; assumes buildSkipFilters called beforehand
+	for _, stream := range cfg.Status.SkippedImagestreams {
+		streamsThatWereSkipped[stream] = true
+		// if something that was skipped (in Status list) is no longer skipped, build list/capture that fact
+		// this should capture reductions in the skipped list, as well as changing entries, but the number of skips
+		// staying the same
+		if _, ok := h.skippedImagestreams[stream]; !ok {
+			unskippedStreams[stream] = true
+			configChangeAtAll = true // set to true in case sizes are equal, but contents have changed
+			configChangeRequireUpsert = true
+			streamChange = true
 		}
 	}
-	if changeInContent {
-		return changeInContent, changeInContent, clearImageImportErrors
+
+	if streamChange {
+		logrus.Printf("SkippedImagestreams changed from %#v to %#v", cfg.Status.SkippedImagestreams, cfg.Spec.SkippedImagestreams)
 	}
 
-	if len(cfg.Spec.SkippedTemplates) != len(cfg.Status.SkippedTemplates) {
-		logrus.Printf("SkippedTemplates changed from %#v to %#v", cfg.Status.SkippedTemplates, cfg.Spec.SkippedTemplates)
-		if len(cfg.Spec.SkippedTemplates) < len(cfg.Status.SkippedTemplates) {
-			return true, true, false
-		}
-		for _, tpl := range cfg.Status.SkippedTemplates {
-			// even if the skipped list has been increased, if a tpl we were skipping
-			// has been removed, we need to upsert; assumes buildSkipFilters called beforehand
-			if _, ok := h.skippedTemplates[tpl]; !ok {
-				return true, true, false
-			}
+	// see if need to update the cfg from the main loop to clear out any prior image import
+	// errors for skipped streams
+	for _, stream := range cfg.Spec.SkippedImagestreams {
+		importErrors := cfg.Condition(v1.ImportImageErrorsExist)
+		beforeError := cfg.NameInReason(importErrors.Reason, stream)
+		importErrors = h.clearStreamFromImportError(stream, cfg.Condition(v1.ImportImageErrorsExist), cfg)
+		afterError := cfg.NameInReason(importErrors.Reason, stream)
```

Comment:
when would it still be there after the clear?  Can it appear multiple times?

Reply:
it is possible that the skipped streams are NOT in the import error list ... if so, I don't want to bother with an update to the config object ... that is why we track unskips and this separately

Reply:
right but in that case beforeError would be false.  When would afterError ever be true, given that you just cleared this imagestream from the errorlist (if i understand the code correctly)

Reply:
ah you want me to change from 

`if beforeError && !afterError {`

to 

`if beforeError {`

If so, yep, I can do that.

Reply:
yeah..and there's no need to actually set/retrieve afterError right? (since it's always going to be false)